### DISCUSSION TITLE
How to add global interceptor for fetch() like Axios

### DISCUSSION THREAD

**Discussion by Trav1sRen** [vercel/next.js#58425] [Help] [answered]
### Summary

Dear community,
May I ask how to add global interceptor on NextJS enhanced fetch() API,
like what we usually do in Axios?
`axios.interceptors.request.use(successHandler, errorHandler)`

**Comment by devraj** [comment, marked as answer]
I have been trying to solve this as well. I can almost swear that when `v14` was released I was able to set things in the root `layout` and it carried through. 

At the moment I am tracking `14.0.3-canary.7` as I require the `allowOrigins` feature for us to make calls to an external API. This also gives me access to `instrumentation`, I tried:

- Overriding `baseURL` in the `register` function
- register interceptors in the `register` function

and neither one of them seem to carry across the lifecycle of execution of any of the server actions.

At them moment all I can think of is wrapping things in utility functions e.g `setupAxios` and having to call that each time I want to use `axios`. 

My use case involves using `orval` to generate an API client for a REST service, it in turn. uses `axios`, I have tried dropping back to just using fetch and have had troubles with orval, not necessarily `fetch`.

Reason for my thought on using fetch directory is because Next builds on the fetch APIs.
